File: app/src/main/java/dev/blaauwendraad/recipe_book/service/model/RecipeSummary.java
package dev.blaauwendraad.recipe_book.service.model;

import jakarta.annotation.Nullable;

public record RecipeSummary(
        Long id,
        String title,
        @Nullable String description,
        @Nullable String imageName,
        Integer numServings,
        PreparationTime preparationTime,
        @Nullable Author author) {}

File: ui/src/models/domain/RecipeSummary.ts
import type { PreparationTime } from "./PreparationTime";

export default interface RecipeSummary {
    id: number;
    title: string;
    description: string;
    imageName: string | null;
    numServings: number;
    preparationTime: PreparationTime;
    author: {
        userId: number;
        username: string;
    };
}
